.github/workflows/test_linux_jax_wheels.yml

```diff
@@ -0,0 +1,219 @@
+    steps:
+      - name: Checkout TheRock
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+      - name: Checkout rocm-jax (plugin + build scripts)
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          path: jax
+          repository: rocm/rocm-jax
+          ref: ${{ inputs.jax_ref }}
+
+      - name: Setup Python
+        uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
+        with:
+          python-version: "3.12"
+
+      - name: Install AWS CLI
+        run: bash ./dockerfiles/install_awscli.sh
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@7474bc4690e29a8392af63c5b98e7449536d5c3a # v4.3.1
+        with:
+          aws-region: us-east-2
+          role-to-assume: arn:aws:iam::692859939525:role/therock-${{ inputs.release_type }}-releases
+
+      - name: Prepare wheelhouse and download JAX wheels
+        shell: bash
+        working-directory: jax
+        run: |
+          set -euo pipefail
+
+          mkdir -p "${WHEELHOUSE_DIR}"
+
+          PY_VER="${{ inputs.python_versions }}"
+          CP="${PY_VER//./}"
+          BASE="s3://${{ env.S3_BUCKET_PY }}/${{ inputs.s3_subdir }}/${{ inputs.amdgpu_family }}/"
+
+          echo "Copying wheels from ${BASE} -> ${WHEELHOUSE_DIR}"
+          echo "Using Python ${PY_VER}, CP=${CP}"
+
+          # Plugin wheel
+          aws s3 cp "${BASE}" "${WHEELHOUSE_DIR}/" \
+            --recursive \
+            --exclude "*" \
+            --include "jax_rocm7_plugin-*-cp${CP}-cp${CP}-manylinux_*_x86_64.whl"
+
+          # PJRT wheel
+          aws s3 cp "${BASE}" "${WHEELHOUSE_DIR}/" \
+            --recursive \
+            --exclude "*" \
+            --include "jax_rocm7_pjrt-*-py3-none-manylinux_*_x86_64.whl"
+          echo "Contents of wheelhouse:"
+          pwd
+          ls -lah "${WHEELHOUSE_DIR}"
+
+      - name: Build JAX Docker image
+        working-directory: jax
+        env:
+          ROCM_VERSION: ${{ inputs.rocm_version }}
+        run: |
+          set -euo pipefail
+
+          # Extract major.minor.patch from ROCM_VERSION
+          # e.g., "7.9.0.dev0+abcdef" -> "7.9.0"
+          # Replacement: \1.\2.\3 reconstructs the version as "major.minor.patch" using the three captured groups.
+          ROCM_SEMVER="$(echo "${ROCM_VERSION}" | sed -E 's/^([0-9]+)\.([0-9]+)\.([0-9]+).*/\1.\2.\3/')"
+          echo "Full ROCm version: ${ROCM_VERSION}"
+          echo "Using semantic ROCm version: ${ROCM_SEMVER}"
+
+          python3 build/ci_build \
+            --rocm-version="${ROCM_SEMVER}" \
+            --therock-path="${{ inputs.tar_url }}" \
+            build_dockers \
+            -f ubu24
+
+          echo "Local images after build:"
+          docker images --format '{{.Repository}}:{{.Tag}} {{.ID}}' | sort
+
+      - name: Checkout JAX tests repo (for extended tests)
+        uses: actions/checkout@v4
```

Comment:
Pin to a specific SHA, see https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies for the rational.